packages/smooth_app/lib/pages/folksonomy/folksonomy_page.dart

```diff
@@ -0,0 +1,187 @@
+                      }
+                    : null,
+                text: appLocalizations.save,
+              ),
+            );
+          },
+        );
+      },
+    );
+  }
+
+  void _showBottomSheet(BuildContext context, AppLocalizations appLocalizations,
+      FolksonomyProvider provider,
+      {required String key, required String value, String? comment}) {
+    showSmoothModalSheet(
+      context: context,
+      builder: (BuildContext context) {
+        return Wrap(
+          children: <Widget>[
+            ListTile(
+              leading: const Icon(Icons.edit),
+              title: Text(appLocalizations.edit_tag),
+              onTap: () {
+                Navigator.pop(context);
+                _showAddEditDialog(context, appLocalizations, provider,
+                    oldKey: key, oldValue: value);
+              },
+            ),
+            ListTile(
+              leading: const Icon(Icons.delete),
+              title: Text(appLocalizations.remove_tag),
+              onTap: () async {
+                Navigator.pop(context);
+                await provider.deleteTag(key);
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    final FolksonomyProvider provider = context.watch<FolksonomyProvider>();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(appLocalizations.product_tags_title),
+      ),
+      body: provider.isLoading
+          ? const Center(child: CircularProgressIndicator.adaptive())
+          : ListView(
+              padding: const EdgeInsets.all(16.0),
+              children: provider.productTags!.entries
+                  .map((MapEntry<String, ProductTag> entry) {
+                return ListTile(
+                  title: Text(
+                    '${entry.key} : ${entry.value.value}',
+                    style: Theme.of(context).textTheme.bodyLarge,
+                  ),
+                  trailing: provider.isAuthorized
+                      ? IconButton(
+                          icon: const Icon(Icons.more_vert),
+                          onPressed: () {
+                            _showBottomSheet(
+                                context, appLocalizations, provider,
+                                key: entry.key,
+                                value: entry.value.value,
+                                comment: entry.value.comment);
+                          },
+                        )
+                      : null,
+                );
+              }).toList(),
+            ),
+      floatingActionButton: provider.isAuthorized
+          ? FloatingActionButton(
+              onPressed: () {
+                _showAddEditDialog(context, appLocalizations, provider);
+              },
+              child: const Icon(Icons.add),
+            )
+          : Container(),
```

Comment:
If you want an empty widget, we have the `EMPTY_WIDGET` constant in the project.